core/benches/ops/async.rs

```diff
@@ -0,0 +1,222 @@
+  ],
+);
+
+#[op2(fast)]
+pub fn op_void() {}
+
+#[op2(fast)]
+pub fn op_make_external() -> *const c_void {
+  std::ptr::null()
+}
+
+#[op2(async)]
+pub async fn op_async_void() {}
+
+#[op2(async)]
+pub async fn op_async_yield() {
+  tokio::task::yield_now().await
+}
+
+#[op2(async(lazy), fast)]
+pub async fn op_async_yield_lazy() {
+  tokio::task::yield_now().await
+}
+
+#[op2(async(lazy), nofast)]
+pub async fn op_async_yield_lazy_nofast() {
+  tokio::task::yield_now().await
+}
+
+#[op2(async(deferred), fast)]
+pub async fn op_async_yield_deferred() {
+  tokio::task::yield_now().await
+}
+
+#[op2(async(deferred), nofast)]
+pub async fn op_async_yield_deferred_nofast() {
+  tokio::task::yield_now().await
+}
+
+#[op2(async(lazy), fast)]
+pub async fn op_async_void_lazy() {}
+
+#[op2(async(lazy), nofast)]
+pub async fn op_async_void_lazy_nofast() {}
+
+#[op2(async(deferred), fast)]
+pub async fn op_async_void_deferred() {}
+
+#[op2(async(deferred), nofast)]
+pub async fn op_async_void_deferred_nofast() {}
+
+fn bench_op(
+  b: &mut Bencher,
+  count: usize,
+  op: &str,
+  arg_count: usize,
+  call: &str,
+) {
+  #[cfg(not(feature = "unsafe_runtime_options"))]
+  unreachable!(
+    "This benchmark must be run with --features=unsafe_runtime_options"
+  );
+
+  let tokio = tokio::runtime::Builder::new_current_thread()
+    .build()
+    .unwrap();
+  let mut runtime = JsRuntime::new(RuntimeOptions {
+    extensions: vec![testing::init_ops_and_esm()],
+    // We need to feature gate this here to prevent IDE errors
+    #[cfg(feature = "unsafe_runtime_options")]
+    unsafe_expose_natives_and_gc: true,
+    ..Default::default()
+  });
+  let err_mapper =
+    |err| generic_error(format!("{op} test failed ({call}): {err:?}"));
+
+  let args = (0..arg_count)
+    .map(|n| format!("arg{n}"))
+    .collect::<Vec<_>>()
+    .join(", ");
+
+  // Prime the optimizer
+  runtime
+    .execute_script(
+      "",
+      FastString::Owned(
+        format!(
+          r"
+async function run() {{
+  const LARGE_STRING_1000000 = '*'.repeat(1000000);
+  const LARGE_STRING_1000 = '*'.repeat(1000);
+  const LARGE_STRING_UTF8_1000000 = '\u1000'.repeat(1000000);
+  const LARGE_STRING_UTF8_1000 = '\u1000'.repeat(1000);
+  const BUFFER = new Uint8Array(1024);
+  const ARRAYBUFFER = new ArrayBuffer(1024);
+  const {{ {op}: op }} = Deno.core.ensureFastOps();
+  const {{ op_make_external }} = Deno.core.ensureFastOps();
+  const EXTERNAL = op_make_external();
+  function {op}({args}) {{
+    op({args});
+  }}
+  let accum = 0;
+  let __index__ = 0;
+  %PrepareFunctionForOptimization({op});
+  {call};
+  %OptimizeFunctionOnNextCall({op});
+  {call};
+
+  async function bench() {{
+    let accum = 0;
+    for (let __index__ = 0; __index__ < {count}; __index__++) {{ {call} }}
+    return accum;
+  }}
+  
+  %PrepareFunctionForOptimization(bench);
+  await bench();
+  %OptimizeFunctionOnNextCall(bench);
+  await bench();
+
+  return bench;
+}}
```

Comment:
How about making this a separate file that uses `include_str!` here? I find it hard for some of the tests to figure out what's going on without syntax highlighting and this one doesn't use any format strings, which makes it easy to move to a separate file.

Reply:
This one does use a format string -- see `{call}`, `{op}`, and `{count}`. I am not sure of a good way to organize this code externally because it's pretty dynamic and we specifically need to generate JS.

I can try some experiments to clean it up a bit.

Reply:
I replaced these with a template approach -- the files are valid JS so they can be formatted w/`deno fmt` and are somewhat readable